lib/omniauth/strategies/openid_connect.rb

```diff
@@ -207,24 +219,61 @@ def access_token
         )
       end
 
-      def validate_access_token!(access_token)
-        verify_id_token! decode_id_token(access_token.id_token) if options.verify_id_token
+      def id_token
+        defined?(@id_token) || begin
+          encoded = access_token.id_token
+          @id_token = encoded ? decode_id_token(encoded) : nil
+        end
+
+        @id_token
+      end
+
+      def perform_backchannel_logout!(plain_token)
+        return fail!(:missing_logout_token) unless plain_token.present?
+        logout_token = decode_logout_token(plain_token)
+
+        if options.verify_logout_token
+          logout_token.verify!(
+            issuer: options.issuer,
+            client_id: client_options.identifier
+          )
+        end
+
+        options.backchannel_logout_callback.call(logout_token)
+      end
+
+
+      def validate_access_token!(_access_token)
+        verify_id_token! if options.verify_id_token
       end
 
-      def verify_id_token!(id_token)
+      def verify_id_token!
         id_token.verify!(
           issuer: options.issuer,
           client_id: client_options.identifier,
           nonce: stored_nonce
         )
       end
 
+      def store_session_id!
+        sid = id_token&.sid
+        return unless sid
+
+        session['oidc_sid'] = sid
```

Comment:
This is outside of the specification so not really defined, but I suggest we prefix this with `omniauth.` as well in line with the oidc state to make it clear that this value originates in the omniauth flow. Perhaps it could be called `omniauth.oidc_sid`.